executor/admin.go

```diff
@@ -163,6 +167,91 @@ func (e *CheckIndexRangeExec) Close() error {
 	return nil
 }
 
+// RestoreTableExec represents a recover table executor.
+// It is built from "admin restore table by job" statement,
+// is used to recover the table that deleted by mistake.
+type RestoreTableExec struct {
+	baseExecutor
+	jobID int64
+}
+
+// Open implements the Executor Open interface.
+func (e *RestoreTableExec) Open(ctx context.Context) error {
+	if err := e.baseExecutor.Open(ctx); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// Next implements the Executor Open interface.
+func (e *RestoreTableExec) Next(ctx context.Context, chk *chunk.Chunk) error {
+	gcEnable, err := admin.CheckGCEnableStatus(e.ctx)
+	if err != nil {
+		return err
+	}
+	if gcEnable {
+		err = admin.DisableGCForRecover(e.ctx)
```

Comment:
We should never disable gc here. Please disable it in the onRestoreTable. If we crash after we disable the gc, we will never have a chance to enable the gc.